In [100]:
# import modules
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [101]:
# Import data
df_usd = pd.read_csv('btc_usd_sep2_nov30.csv')
df_cme = pd.read_csv('btc_cme_sep2_nov30.txt', sep=",")

In [102]:
# print 1st 5 rows of df_usd
df_usd.head()

Date         Open         High          Low        Close  \
0  2018-09-02  7301.250000  7350.580078  7204.890137  7270.049805   
1  2018-09-03  7270.049805  7417.649902  7246.759766  7369.859863   
2  2018-09-04  7369.890137  7401.430176  6694.060059  6705.029785   
3  2018-09-05  6705.060059  6727.189941  6295.109863  6515.419922   
4  2018-09-06  6515.419922  6549.390137  6345.479980  6411.779785   

     Adj Close     Volume  
0  7270.049805  309327793  
1  7369.859863  343762001  
2  6705.029785  797851163  
3  6515.419922  823309044  
4  6411.779785  344487244

In [103]:
# Add Month column
df_usd['Month'] = df_usd["Date"].str.slice(5,7)
df_usd.head()

Date         Open         High          Low        Close  \
0  2018-09-02  7301.250000  7350.580078  7204.890137  7270.049805   
1  2018-09-03  7270.049805  7417.649902  7246.759766  7369.859863   
2  2018-09-04  7369.890137  7401.430176  6694.060059  6705.029785   
3  2018-09-05  6705.060059  6727.189941  6295.109863  6515.419922   
4  2018-09-06  6515.419922  6549.390137  6345.479980  6411.779785   

     Adj Close     Volume Month  
0  7270.049805  309327793    09  
1  7369.859863  343762001    09  
2  6705.029785  797851163    09  
3  6515.419922  823309044    09  
4  6411.779785  344487244    09

In [104]:
# print 1st 5 rows of df_cme
df_cme.head()

Date    Price     Open     High      Low Vol. Change %
0  Nov 30, 2018  3,895.0  4,225.0  4,265.0  3,835.0    -   -7.59%
1  Nov 29, 2018  4,215.0  4,205.0  4,350.0  4,060.0    -    0.24%
2  Nov 28, 2018  4,205.0  3,775.0  4,345.0  3,770.0    -   11.39%
3  Nov 27, 2018  3,775.0  3,650.0  3,775.0  3,550.0    -    4.28%
4  Nov 26, 2018  3,620.0  3,930.0  4,025.0  3,510.0    -   -7.89%

In [105]:
# Create Month column
df_cme['Month'] = df_cme["Date"].str.slice(0,3)
df_cme.head()

Date    Price     Open     High      Low Vol. Change % Month
0  Nov 30, 2018  3,895.0  4,225.0  4,265.0  3,835.0    -   -7.59%   Nov
1  Nov 29, 2018  4,215.0  4,205.0  4,350.0  4,060.0    -    0.24%   Nov
2  Nov 28, 2018  4,205.0  3,775.0  4,345.0  3,770.0    -   11.39%   Nov
3  Nov 27, 2018  3,775.0  3,650.0  3,775.0  3,550.0    -    4.28%   Nov
4  Nov 26, 2018  3,620.0  3,930.0  4,025.0  3,510.0    -   -7.89%   Nov

In [106]:
# Convert the 'Low' and 'High' column from text to float
df_cme["Low"] = [np.float(x.replace(",", "")) for x in df_cme["Low"]]
df_cme["High"] = [np.float(x.replace(",", "")) for x in df_cme["High"]]

# Check new High and Low columns
df_cme.head()

Date    Price     Open    High     Low Vol. Change % Month
0  Nov 30, 2018  3,895.0  4,225.0  4265.0  3835.0    -   -7.59%   Nov
1  Nov 29, 2018  4,215.0  4,205.0  4350.0  4060.0    -    0.24%   Nov
2  Nov 28, 2018  4,205.0  3,775.0  4345.0  3770.0    -   11.39%   Nov
3  Nov 27, 2018  3,775.0  3,650.0  3775.0  3550.0    -    4.28%   Nov
4  Nov 26, 2018  3,620.0  3,930.0  4025.0  3510.0    -   -7.89%   Nov

In [107]:
# Calculate the mean value of the 'Low' and 'High' columns
print("Low, CME")
print(df_cme['Low'].mean())
print(' ')
print("High, CME")
print(df_cme['High'].mean())

Low, CME
5977.76315789
 
High, CME
6200.72368421


In [108]:
# Calculate the mean values for all of the columns in df_usd
print("USD mean values")
print(' ')
print(df_usd.mean())

USD mean values
 
Open         6.176153e+03
High         6.264355e+03
Low          6.041043e+03
Close        6.139058e+03
Adj Close    6.139058e+03
Volume       3.641961e+08
dtype: float64


In [110]:
# Group CME and USD data sets by month to calculate mean values
print("CME")
print(df_cme.groupby('Month').mean())
print(' ')
print("USD")
print(df_usd.groupby('Month').mean())

CME
              High          Low
Month                          
Nov    5470.384615  5179.807692
Oct    6482.200000  6341.000000
Sep    6678.800000  6444.400000
 
USD
              Open         High          Low        Close    Adj Close  \
Month                                                                    
09     6576.233112  6674.758284  6443.960668  6551.819959  6551.819959   
10     6484.464632  6561.385384  6417.326676  6475.810335  6475.810335   
11     5481.096012  5570.600358  5275.271020  5403.302694  5403.302694   

             Volume  
Month                
09     3.586248e+08  
10     2.775047e+08  
11     4.562731e+08  


## Question 1:  What are the mean values for Bitcoin High and Low daily prices for the spot price and the December futures price over a 3 month period ending November 30th ?

   ### Answer: USD-High = 6264, CME-High = 6201; difference of *63*.  USD-Low = 6041, CME-Low = 5978; difference of *63* also.

## Question 2:  What are the differences in mean values grouped by month ?

### Answer: Sep-High: -4, Sep-Low: 0, difference = *-4* (subtractinig CME from USD values)
   ### Oct-High: 79, Oct-Low: 76, difference = *3*
   ### Nov-High: 101, Nov-Low: 95, difference = *6*

## Question 3:  How does the price difference between spot and future change over time ?

### Answer:  __-2__ in Sep, *77.5* in Oct, *98* in Nov (subtracting CME from USD values)